File: TypeScriptAndAngularProjects/angular/solutions/flights8b/src/app/flights-viewer/model/flight.ts
export class Flight {
  number: number;
  origin: string;
  departure: Date;
  destination: string;
  arrival: Date;

  static fromJson(obj: any): Flight {
    const result = new Flight();
    result.number = parseInt(obj.number, 10);
    result.origin = obj.origin;
    result.departure = new Date(obj.departure);
    result.destination = obj.destination;
    result.arrival = new Date(obj.arrival);
    return result;
  }
}

File: SpringProjects/SpringCoreJava/AnnotationBasedExercises/src/main/java/exercise/finish/part3/Flight.java
package exercise.finish.part3;

import java.util.Date;

public class Flight {
    private String origin;
    private String destination;
    private Date departure;

    public Flight(String origin, String destination, Date departure) {
        super();
        this.origin = origin;
        this.destination = destination;
        this.departure = departure;
    }

    public Date getDeparture() {
        return departure;
    }

    public String getDestination() {
        return destination;
    }

    public String getOrigin() {
        return origin;
    }

    public void setDeparture(Date departure) {
        this.departure = departure;
    }

    public void setDestination(String destination) {
        this.destination = destination;
    }

    public void setOrigin(String origin) {
        this.origin = origin;
    }
}
